test/gtest/ucp/test_ucp_wireup.cc

```diff
@@ -304,12 +304,14 @@ void test_ucp_wireup::send_nb(ucp_ep_h ep, size_t length, int repeat,
             reqs.push_back(req);
         }
 
-        ucs_status_t status = ucp_worker_fence(ep->worker);
-        ASSERT_UCS_OK(status);
+        ucp_request_param_t param = {};
+        void *req = ucp_ep_flush_nbx(ep, &param);
+        ASSERT_UCS_PTR_OK(req);
```

Comment:
why changed fence to flush?

Reply:
otherwise it hangs with ud/tcp, because our strong fence is blocking and we need to progress both peers for it to complete

Reply:
this can cause a deadlock with MPI/shmem as well
can we make the fence non-blocking? switch to wireup_ep like we when need to resolve remote ep?

Reply:
why is it a problem? with emulation we need progress on both sides for other operations as well

Reply:
add FIXME comment why it's a flush and not a fence